### PR TITLE
fix(FEC-9160): captions displayed during midroll ad on AVplayer

### PR DESCRIPTION
**Pull request by Yuvalke**
### Description of the Changes

change position of activeCue remove the text tracks from ui

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment 
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/ima.js [hunk 1]
```diff
@@ -1184,6 +1184,27 @@ class Ima extends BasePlugin implements IMiddlewareProvider, IAdsControllerProvi
...
+  /**
+   * On iOS AVPlayer caption stack on video tag, should re-position to hide
+   * @private
+   * @returns {void}
+   * @instance
+   * @memberof Ima
+   */
+  _hideActiveTextTracksOnAVPlayer(): void {
+    const isIOS = this.player.env.os.name === 'iOS';
+    if (isIOS) {
```
Comment (yairans) [inline src/ima.js]:
```suggestion
    if (isIOS && this. playOnMainVideoTag()) {
```

File: src/ima.js [hunk 2]
```diff
@@ -1184,6 +1184,27 @@ class Ima extends BasePlugin implements IMiddlewareProvider, IAdsControllerProvi
...
+      let tracks = this.player.getVideoElement().textTracks;
+      if (tracks.length) {
```
Comment (yairans) [inline src/ima.js]: redundant. the `for` already protects it

**Comment by Yuvalke** [comment]
> But how do you restore them back then?
> If ad is in middle of subtitle time range then it is removed and it never gets restored for that cue when ad finish, right?

I didn't see why it matters, every change of video src change the tracks so it override the tracks that I changed..

File: src/ima.js [hunk 3]
```diff
@@ -1184,6 +1184,25 @@ class Ima extends BasePlugin implements IMiddlewareProvider, IAdsControllerProvi
...
+   * @instance
+   * @memberof Ima
+   */
+  _hideActiveTextTracksOnAVPlayer(): void {
+    const isIOS = this.player.env.os.name === 'iOS';
+    if (isIOS && this.playOnMainVideoTag()) {
+      let tracks = this.player.getVideoElement().textTracks;
+      for (let i = 0; i < tracks.length; i++) {
+        if (tracks[i].mode === 'showing') {
+          tracks[i].activeCues[0].position = 100;
```
Comment (yairans) [inline src/ima.js]: change to loop

**Pull request merged**